nextcord/types/checks.py

```diff
@@ -36,12 +36,10 @@
 MaybeCoro = Union[T, Coro[T]]
 CoroFunc = Callable[..., Coro[Any]]
 ApplicationCheck = Union[
-    Callable[[ClientCog, Interaction], MaybeCoro[bool]], Callable[[Interaction], MaybeCoro[bool]]
-]
-ApplicationHook = Union[
-    Callable[[ClientCog, Interaction], Coro[Any]], Callable[[Interaction], Coro[Any]]
+    Callable[[Cog, Interaction], MaybeCoro[bool]], Callable[[Interaction], MaybeCoro[bool]]
```

Comment:
Things that use Cog should be in TYPE_CHECKING if Cog is imported in TYPE_CHECKING

(Basically the issue from #888 which could be fixed here instead or in addition depending on which is merged first)

Reply:
bca6ff6